tests/platform_tests/conftest.py

```diff
@@ -63,21 +66,49 @@ def bring_up_dut_interfaces(request, duthosts, enum_rand_one_per_hwsku_frontend_
         for port in ports:
             duthost.no_shutdown(ifname=port)
 
-def analyze_syslog(duthost, messages):
+
+def get_state_times(timestamp, state, state_times):
+    time = timestamp.strftime(FMT)
+    state_name = state.split("|")[0].strip()
+    state_status = state.split("|")[1].strip()
+    state_dict = state_times.get(state_name, {"timestamp": {}})
+    timestamps = state_dict.get("timestamp")
+    if state_status in timestamps:
+        state_dict[state_status+" count"] = state_dict.get(state_status+" count", 1) + 1
+    else:
+        # only capture timestamp of first occurence of the entity. Otherwise, just increment the count above.
+        # this is useful in capturing start point. Eg., first neighbor entry, LAG ready, etc.
+        timestamps[state_status] = time
+    return {state_name: state_dict}
+
+
+def analyze_syslog(duthost, messages, result, first_occurence_times):
+    service_restart_times = dict()
     if not messages:
         logging.error("Expected messages not found in syslog")
         return None
 
-    service_restart_times = dict()
+    reboot_pattern = re.compile(r'.* NOTICE admin: Rebooting with /sbin/kexec -e to.*...')
     service_patterns = {
         "Stopping": re.compile(r'.*Stopping.*service.*'),
         "Stopped": re.compile(r'.*Stopped.*service.*'),
         "Starting": re.compile(r'.*Starting.*service.*'),
         "Started": re.compile(r'.*Started.*service.*')
     }
+    other_patterns = {
+        "PORT_INIT|Start": re.compile(r'.*swss#portsyncd.*main.*PortInitDone.*'),
+        "LAG_READY|Start": re.compile(r'.*teamd#tlm_teamd.*try_add_lag.*The LAG \'PortChannel.*\' has been added.*'),
+        "INIT_VIEW|Start": re.compile(r'.*swss#orchagent.*notifySyncd.*sending syncd.*INIT_VIEW.*'),
+        "INIT_VIEW|End": re.compile(r'.*swss#orchagent.*sai_redis_notify_syncd.*switched ASIC to INIT VIEW.*'),
+        "APPLY_VIEW|Start": re.compile(r'.*swss#orchagent.*notifySyncd.*sending syncd.*APPLY_VIEW.*'),
+        "APPLY_VIEW|End": re.compile(r'.*swss#orchagent.*sai_redis_notify_syncd.*switched ASIC to APPLY VIEW.*'),
+        "FINALIZER|Start":  re.compile(r'.*WARMBOOT_FINALIZER.*Wait for database to become ready.*'),
+        "FINALIZER|End":  re.compile(r'.*WARMBOOT_FINALIZER.*Finalizing warmboot.*')
+    }
 
     def service_time_check(message, status):
-        time = message.split(duthost.hostname)[0].strip()
+        time = datetime.strptime(message.split(duthost.hostname)[0].strip(), FMT)
+        time = time.strftime(FMT)
```

Comment:
I feel this might be redundant. Seems like we convert a string to datetime assuming the format is `FMT`, then we convert it back to string with the same `FMT`. So shouldn't these two lines equivalent to `time = message.split(duthost.hostname)[0].strip()`?

Reply:
You are right, these two statements _seem_ equivalent to just striping the time from the message.
Except that the `strftime` call adds an extra digit to date (`May 5` vs `May 05`). I added this ONLY so that timing signature from syslog (1 digit) and sairedis (2 digits) matches and uploaded data in Kusto is in same format. 

```
>>> 
>>> message = "May  5 19:55:03.125650 vlab-01 NOTICE swss#portsyncd: :- main: PortInitDone"
>>> time = datetime.strptime(message.split("vlab-01")[0].strip(), FMT)                     
>>> time = time.strftime(FMT)
>>> time
'May 05 19:55:03.125650'
>>> 
```
If only time was stripped from the message, date will have only one digit (and the format will look differently from the date collected from sairedis.
```
>>> message.split("vlab-01")[0].strip()       
'May  5 19:55:03.125650'
>>> 
```